init.lua

```diff
@@ -1,140 +1,153 @@
-    '1 got too close to the camp fire',
-    '1 got nuked'
+    S('1 burned to a crisp'),
+    S('1 got a little too warm'),
+    S('1 got too close to the camp fire'),
+    S('1 got nuked')
 }
 
 -- Drowning
 drown = {
-    '1 drowned',
-    '1 ran out of air',
-    '1 tried to impersonate an anchor',
-    "1 forgot they weren't a fish"
+    S('1 drowned'),
+    S('1 ran out of air'),
+    S('1 tried to impersonate an anchor'),
+    S("1 forgot they weren't a fish")
 }
 
 -- Burning in lava
 lava = {
-    '1 thought lava was cool',
-    '1 tried to swim in lava',
-    '1 melted in lava',
-    '1 fell whilst trying to reach that last diamond'
+    S('1 thought lava was cool'),
+    S('1 tried to swim in lava'),
+    S('1 melted in lava'),
+    S('1 fell whilst trying to reach that last diamond')
 }
 
 -- Killed by other player
 pvp = {
-    '1 was slain by 2',
-    '1 was killed by 2',
-    '1 was put to the sword by 2',
-    '1 lost a PVP battle to 2'
+    S('1 was slain by 2'),
+    S('1 was killed by 2'),
+    S('1 was put to the sword by 2'),
+    S('1 lost a PVP battle to 2')
 }
 
 -- Killed by mob
 mob = {
-    '1 was slain by 2',
-    '1 was killed by 2',
-    "1 got on 2's last nerve",
-    '1 forgot to feed 2'
+    S('1 was slain by 2'),
+    S('1 was killed by 2'),
+    S("1 got on 2's last nerve"),
+    S('1 forgot to feed 2')
 }
 
 -- Everything else
 other = {
-    '1 died',
-    '1 gave up on life',
-    '1 passed out - permanantly',
-    '1 did something fatal'
+    S('1 died'),
+    S('1 gave up on life'),
+    S('1 passed out - permanantly'),
+    S('1 did something fatal')
 }
 
-function send_death_message(cause, victim, killer)
+function get_death_message(cause)
+    return cause[math.random(1, #cause)]
+end
+
+function send_death_message(message, victim, killer)
     meta = victim:get_meta()
     show_death_messages = meta:get_string('show_death_messages')
 
     if show_death_messages == '' or show_death_messages == 'yes' then
-        random_selection = cause[math.random(4)]
-        death_message = string.gsub(random_selection, '1', victim:get_player_name())
-
-        if killer then
-            if killer:is_player() then
-                death_message = string.gsub(death_message, '2', killer:get_player_name())
-            else
-                -- Get entity name, excluding mod name ("mymod:enemy" -> "enemy")
-                entity_name = killer:get_luaentity().name
-                index, _ = string.find(entity_name, ':')
-                entity_name = string.sub(entity_name, index + 1)
-
-                death_message = string.gsub(death_message, '2', entity_name)
+        for _, player in ipairs(minetest.get_connected_players()) do
+            local ricevee_name = player:get_player_name()
+            local info = minetest.get_player_information(ricevee_name)
+            local lang = (info and info.lang_code) or "en"
+            local death_message = minetest.get_translated_string(lang, message)
```

Comment:
Looping through each player, translating it according to their locale, and sending them their translated message is not necessary or efficient. See [the Minetest docs](https://github.com/minetest/minetest/blob/master/doc/lua_api.md#operations-on-translated-strings).

Reply:
Thanks, just force-pushed a fix to do this client-side instead.